dumper/app/src/main/sh/java_versioning.sh

```diff
@@ -1,16 +1,28 @@
 #!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 function is_java_greater_than_8() {
-  java_path=$(which java)
-  java_version_output=$($java_path -version 2>&1)
+  java_version_output="$(java -version 2>&1)"
```

Comment:
Not all JDKs support java -version. Some of them only support javac -version. I _think_ that's a non-issue post-8, but it's worth a check, particularly on the IBM JDK we ran into the other day (J9) as we clearly have customers running some really odd Java versions.

Reply:
After merging this PR, this script (i.e. `java_versioning.sh`) will no longer be used by the dumper, only by GCE launcher. I merely added a license header and applied some small fixes, but this doesn't change the script behavior.

I would suggest to apply any changes to this script in a separate PR that would focus on GCE launcher. One of the options is that we can remove this script and use the same trick from this PR (i.e. the `JDK_JAVA_OPTIONS` variable) in the GCE launcher start scripts.